File: Python Course/Software Development Module - in Python/Week11/HairdresserApp.py
"""
HairdresserApp.py
Billy Cussen
03/10/2020
"""

#Objects
import gc
import Hairdresser 
from Hairdresser import myAppointment

#Variables
repeat = "y"
inputCheck = True

while repeat == "y":

    #Input - Appointment Type
    while inputCheck:
        try:
            appointmentType = str(input("Please enter your Appointment Type (\"Style\" or \"Cut\"): "))
            appointmentType = appointmentType.lower()
        except:
            print("You need to enter your appointment type!")
        else:
            if(appointmentType != "style" and appointmentType != "cut"):
                print("You need to enter \"Style\" or \"Cut\"!")
            else:
                inputCheck = False

    #Input - Hair Type
    while inputCheck == False:
        try:
            hairType = str(input("Please enter your Hair Type (\"Long\", \"Mid\" or \"Short\"): "))
            hairType = hairType.lower()
        except:
            print("You need to enter your Hair type!")
        else:
            if(hairType != "long" and hairType != "mid" and hairType != "short"):
                print("You need to enter \"Long\", \"Mid\" or \"Short\"!")
            else:
                inputCheck = True

    #Process & Output
    try:
        myAppointmentApp = myAppointment(appointmentType, hairType)
        myAppointmentApp.computeAppointmentDay()
        print(myAppointmentApp.day)
    except Exception as e:
        print(str(e))
    finally:
        myAppointmentApp = None
        gc.collect()

    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"Y\" to continue or \"N\" to quit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat !="n":
            print("You need to enter \"y\" or \"n\"")
    
print("Thanks for using my App!")

File: Java/Software Development Module - Java/Week11/HairdresserApp.java
/*
HairdresserApp.java
Billy Cussen
21/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class HairdresserApp {
    
    public static void main(String[]args){

        //Variables
        String appointmentType = new String();
        String hairType = new String();
        String appointmentDay;
        boolean inputCheck;

        //Object
        Scanner keyboard;
        Hairdresser myHairdresser;

        do{
            keyboard = new Scanner(System.in);
            inputCheck = true;

            try{
                System.out.print("Please enter your Appointment Type (Style or Cut): ");
                appointmentType = keyboard.nextLine();

                if(!(appointmentType.equalsIgnoreCase("Style")||appointmentType.equals("Cut"))){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                inputCheck = false;
                System.out.println("Please enter Style or Cut");
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        do{
            keyboard = new Scanner(System.in);
            inputCheck = true;

            try{
                System.out.print("Please enter your Hair Type (Long, Mid or Short): ");
                hairType = keyboard.nextLine();

                if(!(hairType.equalsIgnoreCase("Long")||hairType.equals("Mid")||hairType.equals("Short"))){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                inputCheck = false;
                System.out.println("Please enter Long, Mid or Short");
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        myHairdresser = new Hairdresser(appointmentType,hairType);
        appointmentDay = myHairdresser.computeAndGetAppointmentDay();
        myHairdresser = null;
        System.out.println(appointmentDay);
    }

}